### ISSUE TITLE
BMS Altimeter - If QNH too low -> Wrong Indicated Altitute

### ISSUE THREAD

**Issue by andreashohl82** [HeliosVirtualCockpit/Helios#389]
Hello again,

by mistake I stumbled accross an issue with the altimeter:
I you reduce the barometric pressure too much then the indicated altitude is not correct.
i.e.: While standing on the ground reduce the QNH down to read an altitude of 99900 ft. in BMS. Helios shows 100 ft. which is not correct. It seems that Helios only knows "positive" altitudes.

Tested with Version: 1.6.3425.0

**Comment by Todd1215** [comment]
If you are talking about indicated altitude which is tied to calibration then yes it only outputs positive values due to using abs  function Math.Abs(_lastFlightData2.aauz)

I guess the question is could an altitude with barometric calibration cause a negative value. To me a negative altitude means your dead...but I'm just an avid flight simmer not an actual pilot so I could be wrong in my assumption.

**Comment by andreashohl82** [comment]
Well, not allways.
While you fly above the transittion level then you have the altimiter set to 29.92 (or 1013). If you descend into a target area where the local pressure is not known and you have to go low (due to SAM coverage or other circumstances) then it could happen that the altitude becomes "negative" due to a low local pressure. If the pilot looks at the altimeter and sees i.e. 200 ft. then it should mean 200 ft. and not something much lower. If the altimeter would show 99800 (= -200 ft.) then the pilot knows that he has to disregard the altimeter in this situation.
To sum it up: Since BMS is a combat flight simulation there may be situations where it is crutial to also read "negative" altitudes :)

**Comment by Todd1215** [comment]
Looking at the code it looks like it changed in 1.4 to abs value which may not be correct in all situations.

**Comment by Todd1215** [comment]
Just checked the value of 'indicated altitude' and if you go below the current pressure setting things start to go wacky. Some of this is caused by the Altimeter Gauge that is in Helios. I'm not even sure the gauge can represent the needle moving backwards with negative values. Needs some investigation to figure this one out.

**Comment by Todd1215** [comment]
This code change may help with this issue. Fixed by commit 99d901b

**Comment by andreashohl82** [comment]
That's not a big issue. I already have a custom altimeter in place where the drum numbers rotate from top to down as it is that way in BMS.

**Comment by andreashohl82** [comment]
I compiled the current Master and the issue does still exist. Was the code change not merged into the Master branch?
